### PR TITLE
improvements to ignore-missing option

### PR DESCRIPTION
**Pull request by kthhrv**
previous attempt was incomplete, I've tested this behaves as expect on an ec2 instance for which 404 is retruned

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -9,6 +9,8 @@ Pending Release
 
 * (Insert new release notes below this line)
 
+* Add `--ignore-missing` argument to `exec` command to allow it to run when user-data isn't configured for the current EC2 instance
```
Comment (adamchainz) [inline HISTORY.rst]: wrap at 80 chars

File: treehugger/ec2.py [hunk 2]
```diff
@@ -25,8 +25,11 @@ def load_user_data_as_yaml_or_die(ignore_missing=False):
         die('Could not connect to EC2 metadata service - are we on an EC2 instance?')
         raise SystemExit(1)
     except HTTPError as exc:
-        if not ignore_missing:
-            die('Got a {} from the EC2 metadata service when retrieving user data'.format(exc.response.status_code))
+        if exc.response.status_code == 404 and ignore_missing:
+            return {'TREEHUGGER_APP': 'Missing', 'TREEHUGGER_STAGE': 'Missing'}
+
+        die('Got a {} from the EC2 metadata service when retrieving user data'.format(exc.response.status_code))
+        raise SystemExit(1)
```
Comment (adamchainz) [inline treehugger/ec2.py]: `raise` is unnecessary as `die` does the exit for us. Line 26 can be removed too.

**Comment by adamchainz** [comment]
`test_exec` uses `responses` (https://github.com/getsentry/responses) and could easily be adapted with a 404 to check the mocked `execlp` is called.

File: treehugger/ec2.py (hunk 2 above)
Comment (kthhrv) [reply treehugger/ec2.py]: 26 and 32 gone

File: HISTORY.rst (hunk 1 above)
Comment (kthhrv) [reply HISTORY.rst]: done

**Comment by kthhrv** [comment]
test added

File: tests/test_cli.py [hunk 3]
```diff
@@ -311,6 +311,23 @@ def test_exec(self, kms_stub, capsys):
             'TREEHUGGER_STAGE': 'qux',
         }
 
+    @responses.activate
+    def test_exec_with_ignore_missing(self, kms_stub, capsys):
+        responses.add(
+            responses.GET,
+            USER_DATA_URL,
+            status=404,
+        )
```
Comment (adamchainz) [inline tests/test_cli.py]: could be one line

Comment (kthhrv) [reply tests/test_cli.py]: done

**Pull request merged**